qiita_ware/metadata_pipeline.py

```diff
@@ -0,0 +1,87 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+
+from future.utils import viewvalues, viewkeys
+
+from qiita_db.metadata_template import (load_template_to_dataframe,
+                                        SampleTemplate, PrepTemplate,
+                                        PREP_TEMPLATE_COLUMNS,
+                                        PREP_TEMPLATE_COLUMNS_TARGET_GENE,
+                                        CONTROLLED_COLS,
+                                        TARGET_GENE_DATA_TYPES)
+from qiita_db.util import convert_from_id
+from qiita_ware.exceptions import QiitaWareError
+
+
+def create_templates_from_qiime_mapping_file(fp, study, data_type):
+    """Creates a sample template and a prep template from qiime mapping file
+
+    Parameters
+    ----------
+    fp : str or file-like object
+        Path to the QIIME mapping file
+    study : Study
+        The study to which the sample template belongs to
+    data_type : str or int
+        The data_type of the prep_template
+
+    Returns
+    -------
+    (SampleTemplate, PrepTemplate)
+        The templates created from the QIIME mapping file
+    """
+    qiime_map = load_template_to_dataframe(fp, index='#SampleID')
+
+    # There are a few columns in the QIIME mapping file that are special and
+    # we know how to deal with them
+    rename_cols = {
+        'BarcodeSequence': 'barcode',
+        'LinkerPrimerSequence': 'primer',
+        'Description': 'description',
+    }
+
+    if 'ReverseLinkerPrimer' in qiime_map:
+        rename_cols['ReverseLinkerPrimer'] = 'reverselinkerprimer'
+
+    missing = set(rename_cols).difference(qiime_map.columns)
+    if missing:
+        raise QiitaWareError(
```

Comment:
Can you add this exception to this [try/except](https://github.com/biocore/qiita/blob/master/qiita_pet/handlers/study_handlers/description_handlers.py#L192).